types/src/binary_port/binary_response.rs

```diff
@@ -0,0 +1,179 @@
+//! The binary response.
+
+use crate::{
+    bytesrepr::{self, Bytes, FromBytes, ToBytes},
+    ProtocolVersion,
+};
+use alloc::vec::Vec;
+
+#[cfg(test)]
+use crate::testing::TestRng;
+
+use super::{
+    binary_response_header::BinaryResponseHeader,
+    db_id::DbId,
+    payload_type::{PayloadEntity, PayloadType},
+    DbRawBytesSpec, ErrorCode,
+};
+
+/// The response use in the binary port protocol.
+#[derive(Debug, PartialEq)]
+pub struct BinaryResponse {
+    /// Header of the binary response.
+    header: BinaryResponseHeader,
+    /// The response.
+    payload: Vec<u8>,
+}
+
+impl BinaryResponse {
+    /// Creates new empty binary response.
+    pub fn new_empty(protocol_version: ProtocolVersion) -> Self {
+        Self {
+            header: BinaryResponseHeader::new(None, protocol_version),
+            payload: vec![],
+        }
+    }
+
+    /// Creates new binary response with error code.
+    pub fn new_error(error: ErrorCode, protocol_version: ProtocolVersion) -> Self {
```

Comment:
I think the errors should contain more info than just the error code.  There should be functionality to provide an error message which should ultimately be returned to the client of the sidecar in most/all cases.

Reply:
nothing prevents us from adding `PayloadType`s for errors, which means we could just add an error struct with a message in it that could be optionally attacahed to errors, we just didn't deem it necessary for the existing RPC API

Reply:
Yes, similarly to [this](https://github.com/casper-network/casper-node/pull/4491#discussion_r1461572730), we might consider doing a more comprehensive error handling/reporting via binary port, but most likely as a follow-up PR or even a separate feature.